src/WarcraftLegacies.Source/Quests/Cthun/QuestRebuildAhnqiraj.cs

```diff
@@ -22,15 +24,28 @@ public QuestRebuildAhnqiraj(Rectangle questRect, unit gateAhnQiraj) : base("Rebu
       @"ReplaceableTextures\CommandButtons\BTNCthunHatchery.blp")
     {
       _gateAhnQiraj = gateAhnQiraj;
-      AddObjective(new ObjectiveBuildUniqueBuildingsInRect(questRect, "in outer Ahn'Qiraj", 5));
+      _gateAhnQiraj.SetInvulnerable(true);
```

Comment:
Move to an `OnAdd` method override. Trying to move away from constructors having an impact on the game state.